SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestNetReference/SpatialTestNetReference.cpp

```diff
@@ -0,0 +1,207 @@
+		CubeLocations.Add(FVector(0.0f, 10000.0f, 40.0f));
+		CubeLocations.Add(FVector(-10000.0f, 0.0f, 40.0f));
+
+		// Spawn the cubes
+		TArray<ACubeWithReferences*> TestCubes;
+		int NumberOfCubes = CubeLocations.Num();
+
+		for (int i = 0; i < NumberOfCubes; ++i)
+		{
+			ACubeWithReferences* CubeWithReferences = NetTest->GetWorld()->SpawnActor<ACubeWithReferences>(CubeLocations[i], FRotator::ZeroRotator, FActorSpawnParameters());
+			TestCubes.Add(CubeWithReferences);
+		}
+
+		// Set the cubes' references
+		for (int i = 0; i < NumberOfCubes; ++i)
+		{
+			TestCubes[i]->Neighbour1 = TestCubes[(i + 1) % NumberOfCubes];
+			TestCubes[i]->Neighbour2 = TestCubes[(i + NumberOfCubes -1) % NumberOfCubes];
+		}
+
+		// Set the PositionUpdateFrequency to a higher value so that the amount of waiting time before checking the references can be smaller, decreasing the overall duration of the test
+		Test->PreviousPositionUpdateFrequency = GetMutableDefault<USpatialGDKSettings>()->PositionUpdateFrequency;
+		GetMutableDefault<USpatialGDKSettings>()->PositionUpdateFrequency = 10000.0f;
+
+		// Spawn the TestMovementCharacter actor for client 1 to possess.
+		for (ASpatialFunctionalTestFlowController* FlowController : Test->GetFlowControllers())
+		{
+			if (FlowController->ControllerType == ESpatialFunctionalTestFlowControllerType::Client && FlowController->ControllerInstanceId == 1)
+			{
+				ATestMovementCharacter* TestCharacter = Test->GetWorld()->SpawnActor<ATestMovementCharacter>(FVector::ZeroVector, FRotator::ZeroRotator, FActorSpawnParameters());
+				AController* PlayerController = Cast<AController>(FlowController->GetOwner());
+
+				Test->OriginalPawn = TPair<AController*, APawn*>(PlayerController, PlayerController->GetPawn());
+				Test->RegisterAutoDestroyActor(TestCharacter);
+
+				PlayerController->Possess(TestCharacter);
+			}
+		}
+
+		Test->FinishStep();
+		});
+
+	AddClientStep(TEXT("SpatialTestNetReferenceClientExecuteTest"), 1, nullptr, nullptr, [](ASpatialFunctionalTest* NetTest, float DeltaTime) {
+		ASpatialTestNetReference* Test = Cast<ASpatialTestNetReference>(NetTest);
+		AController* PlayerController = Cast<AController>(NetTest->GetLocalFlowController()->GetOwner());
+		ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+
+		// The index of the current move, the % is required since the test goes through each test location twice.
+		int CurrentMoveIndex = Test->LocationIndex % Test->TestLocations.Num();
+
+		// Move the character to the correct location
+		if(Test->TimerHelper == 0.0f)
+		{
+			PlayerCharacter->ServerMoveToLocation(Test->TestLocations[CurrentMoveIndex].Key);
+		}
+
+		// After arriving at the correct location, perform the required checks
+		if (PlayerCharacter->GetActorLocation().Equals(Test->TestLocations[CurrentMoveIndex].Key, 1.0f))
+		{
+			// Wait for a second as it may take some time for the references to be updated correctly
+			if (Test->TimerHelper < 0.1f)
```

Comment:
I think instead of being a timer to wait a certain amount of time and then check, it should instead check every frame up to some timeout when we assume it's wrong.